### PR TITLE
Add customer support link in the short footer

### LINKED ISSUE #39047: Add the link to IT/Customer Support
The theme settings allow to set up two additional links, which a user may use: Legal Notice and Privacy. I suggest to make them either adjustable or add the third one: IT or Customer Support, which a user may click and get access to a special webpage or resource, where he/she may get appropriate support (in my case that is the Signal Group).
image
image

### PR DESCRIPTION
**Pull request by aaygupta**
* Resolves: #39047 

## Summary
Add customer support link in the short footer, add translations for en_GB, update tests.
image

## TODO
- Update Documentation
- Add Translations in other languages

### CODE REVIEW

**Comment by kesselb** [comment]
Hi @aaygupta :wave: 

Thanks for your great contribution :+1: 

- Please revert the changes to the translations. We use transifx to manage translations. A background job will extract the source strings, push them to transifex and pull back the translations later. 
- Please commit the changes to `dist/theming-admin-theming.js` and `dist/theming-admin-theming.js.map`

File: apps/theming/lib/ThemingDefaults.php [hunk 1]
```diff
@@ -157,6 +157,10 @@ public function getSlogan(?string $lang = null) {
 		return \OCP\Util::sanitizeHTML($this->config->getAppValue('theming', 'slogan', parent::getSlogan($lang)));
 	}
 
+	public function getSupportUrl() {
```
Comment (kesselb) [inline apps/theming/lib/ThemingDefaults.php:160]:
```suggestion
	public function getSupportUrl(): string {
```

File: apps/theming/lib/ThemingDefaults.php [hunk 2]
```diff
@@ -157,6 +157,10 @@ public function getSlogan(?string $lang = null) {
...
+		return (string)$this->config->getAppValue('theming', 'supportUrl', '');
```
Comment (kesselb) [inline apps/theming/lib/ThemingDefaults.php:161]:
```suggestion
		return $this->config->getAppValue('theming', 'supportUrl', '');
```

File: apps/theming/src/AdminTheming.vue [hunk 3]
```diff
@@ -136,6 +136,7 @@ const {
 	slogan,
 	url,
 	userThemingDisabled,
+  customerSupportUrl,
```
Comment (kesselb) [inline apps/theming/src/AdminTheming.vue:139]:
```suggestion
	customerSupportUrl,
```

Comment (kesselb) [reply apps/theming/src/AdminTheming.vue:139]: Please indent with one tab.

**Comment by aaygupta** [comment]
Hi @kesselb, all the mentioned changes have been made. 

Regards
Ankit Gupta

File: apps/theming/lib/ThemingDefaults.php [hunk 4]
```diff
@@ -181,6 +185,10 @@ public function getShortFooter() {
 		$footer .= ($slogan !== '' ? ' – ' . $slogan : '');
 
 		$links = [
+			[
+				'text' => $this->l->t('Customer support'),
+				'url' => (string)$this->getSupportUrl()
```
Comment (kesselb) [inline apps/theming/lib/ThemingDefaults.php:190]:
```suggestion
				'url' => $this->getSupportUrl()
```

**Review by kesselb — APPROVED** [review]

**Comment by aaygupta** [comment]
Hi @kesselb  I have rebased the branch.

**Review by Altahrim — APPROVED** [review]

**Comment by szaimen** [comment]
Conflicts...

**Comment by Altahrim** [comment]
@aaygupta Can you update the PR please?
You'll have to rebase, fix conflicts and use [conventional commits](https://www.conventionalcommits.org/)

**Pull request closed without merging**